internal/pkg/service/common/dependencies/encryption.go

```diff
@@ -0,0 +1,47 @@
+package dependencies
+
+import (
+	"context"
+
+	"github.com/keboola/go-cloud-encrypt/pkg/cloudencrypt"
+
+	"github.com/keboola/keboola-as-code/internal/pkg/service/stream/encryption"
+	"github.com/keboola/keboola-as-code/internal/pkg/utils/errors"
+)
+
+// encryptionScope implements EncryptionScope interface.
+type encryptionScope struct {
+	encryptor cloudencrypt.Encryptor
+}
+
+type encryptionScopeDeps interface {
+	BaseScope
+	EtcdClientScope
+}
+
+func NewEncryptionScope(ctx context.Context, cfg encryption.Config, d encryptionScopeDeps) (EncryptionScope, error) {
+	return newEncryptionScope(ctx, cfg, d)
+}
+
+func newEncryptionScope(ctx context.Context, cfg encryption.Config, d encryptionScopeDeps) (v *encryptionScope, err error) {
+	ctx, span := d.Telemetry().Tracer().Start(ctx, "keboola.go.common.dependencies.NewEncryptionScope")
```

Comment:
```suggestion
	ctx, span := d.Telemetry().Tracer().Start(ctx, "keboola.go.common.dependencies.EncryptionScope")
```

Reply:
![image](https://github.com/user-attachments/assets/09f44b4a-c935-4ce0-ae27-82d018f8dd0b)

Reply:
I don't like the `New` prefix. Maybe create jira task for that. Why do we want to trace it like that ?

Reply:
I believe these spans are just for the creation of the dependencies so it kinda makes sense to me? Anyway this was set up by Michal so I don't really know the answers.